File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.cpp
// Sol1 - gpt4

#include <vector>
#include <algorithm>

class Solution {
public:
    int largestAltitude(std::vector<int>& gain) {
        int altitude = 0;  // Current altitude, starts at 0
        int max_altitude = 0;  // Maximum altitude, initialized with 0

        // Iterate through the gain array
        for (int g : gain) {
            altitude += g;  // Add the gain to the current altitude
            max_altitude = std::max(max_altitude, altitude);  // Update the maximum altitude
        }

        return max_altitude;
    }
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.py
# Sol1 - GPT4

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitude = 0  # Current altitude, starts at 0
        max_altitude = 0  # Maximum altitude, initialized with 0

        # Iterate through the gain array
        for g in gain:
            altitude += g  # Add the gain to the current altitude
            max_altitude = max(max_altitude, altitude)  # Update the maximum altitude

        return max_altitude
